resources/build/build-utils.md

```diff
@@ -339,6 +360,29 @@ this, you should use `--debug,-d` to enable the shorthand form.
 Note that you should not include any of the [standard builder parameters] here.
 
 
+**Dependencies** are defined with a `@` prefix, for example:
+
+```bash
+builder_describe "Sample script" \
+  "@/core configure build" \
+  configure \
+  build
+```
+
+A dependency always starts with `@`. The path to the dependency will be relative
+to the build script folder, or to the root of the repository, if the path starts
+with `/`, not to the root of the file system. It is an error to specify a
+dependency outside the repo root.
+
+Relative paths will be expanded to full paths, again, relative to the root of
+the repository.
+
+Dependencies may be limited to specific `action:target` pairs on the current
+script. If not specified, dependencies will be built for all actions on all
+targets. Either `action` or `:target` may be omitted, and multiple actions and
+targets may be specified, space separated.
+
+
```

Comment:
Since we're enhancing the docs, I think it'd be good to document this tidbit explicitly; if my understanding is correct, this is an up-until-now undocumented limitation of the current implementation.

```suggestion
Dependency builds will always use only the default option and will
never specify a specific target.  There is currently no mechanism
provided to configure the call structure for dependency builds.

```

Reply:
A clear case of this:  Keyman for Android depends on Web's `:embed` target.  It couldn't care less about the `:dom` / `:web` target, which integrates into websites.  If "depending" on Web's main `build.sh`, Android has no choice but to build both.

Of course, for this _specific_ case, Android could depend on a `web/src/app/embed/build.sh` instead, which would work fine.

Reply:
Yes, even better, Android could depend on `/common/web/app-embed/build.sh` (or something like that, `app-embed` or `app-host`, because `embed` is beautifully ambiguous, given we 'embed' KeymanWeb into web pages also...). Then it becomes separated out from `/web/build.sh`, which targets _only_ browser-based usage.

This also makes it easy to determine which modules of web belong in `/common/web`: any module which `app-host` depends on. Those that are used only by `/web` would stick around in `/web`.

Reply:
The principle here is that dependencies should be simple. The ideal build.sh script has a single target (apart from parent/child scripts -- which we are discussing shortly).

Reply:
Note that this, in turn, could probably use some formal documentation - these details have been _implied_ until now, but it takes significant inference to come to these conclusions.  Note that I only got halfway there with my comment, and I've been using the dependency system for a while.

Making this aspect of the design more explicit would definitely help as more people become accustomed to the dependency system.

Reply:
Possible other name: `webview-engine` or similar, as  the current `:embed` is designed to be managed by an app-hosted webview as its core engine.  Pretty sure both Android and iOS use the term WebView for part of the app's keyboard's host container.

Reply:
> Note that this, in turn, could probably use some formal documentation - these details have been implied until now, but it takes significant inference to come to these conclusions. Note that I only got halfway there with my comment, and I've been using the dependency system for a while.

> Making this aspect of the design more explicit would definitely help as more people become accustomed to the dependency system.

Happy to hear suggestions on how this could be clearly documented here.